.github/workflows/makefile.yaml

```
@@ -0,0 +1,63 @@
+name: Makefile
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      image: docker://ghcr.io/pulumi/devcontainer:latest
+      options: --user root
+    permissions:
+      contents: read
+      packages: read
+      actions: read
+    steps:
+    -
+      name: Checkout repository
+      uses: actions/checkout@v4
+      id: git
+      with:
+        submodules: 'recursive'
+    -
+      name: Unshallow clone for tags
+      id: tags
+      run: |
+        set -ex
+        sudo chown -R $(whoami) /__w/pulumi-provider-boilerplate/pulumi-provider-boilerplate
+        git config --global --add safe.directory /__w/pulumi-provider-boilerplate/pulumi-provider-boilerplate
+        git fetch --prune --unshallow --tags
+    -
+      name: Build
+      id: build
+      run: |
+        set -ex
+        git config --global --add safe.directory /__w/pulumi-provider-boilerplate/pulumi-provider-boilerplate
+        make build
+    -
+      name: Install
+      id: install
+      run: |
+        set -ex
+        make install
+    -
+      name: PulumiUp
+      id: deploy
+      run: |
+        set -ex
+        make deploy
```

We don't need `set -ex` for one liners.
```suggestion
      run: make deploy
```